### ISSUE TITLE
getHost doesn't identify non HTTP(S) URLs as remote

### ISSUE THREAD

**Issue by yaronyg** [pouchdb/pouchdb#1602]
_tl;dr - New URL types handled via an adapter are not marked as remote by getHost() causing the URL to be deformed when processed by getDBURL()_

In lib/adapters/http.js/getHost() the function tests an incoming URL with:

``` Javascript
  if (/http(s?):/.test(name)) {
```

I am using an adapter that supports a new URL mechanism (httpkey) which now works thanks to the support y'all put in for pluggable xhr (which, btw, works great!!).

The problem is that a httpkey URL fails this test and so its described by the last part of getHost()

``` Javascript
return {host: '', path: '/', db: name, auth: false};
```

So the input URL was: httpkey://127.0.0.1:9898/rsapublickey:0.0/test
The output becomes:

``` Json
{
 "auth" : "false",
 "db" : "httpkey://127.0.0.1:9898/rsapublickey:0.0/test",
 "host" : "",
 "path" : "/"
}
```

Both host and path are wrong and the URL is not marked as remote. All of this comes back to haunt us in the call to getDBURL(). That code checks to see if the URL is remote. In my case it will not be detected as remote because of the previous code in getHost(). So this means that getDBURL() will run the following code (note that path variable below is different than the one in the object above):

``` Javascript
  return '/' + opts.db + '/' + path; 
```

So the original URL:
httpkey://127.0.0.1:9898/rsapublickey:0.0/test 
gets turned into something like:
/httpkey://127.0.0.1:9898/rsapublickey:0.0/test/

The URL, in this case, is wrapped with "/" characters at the start and end. 

Now XHR could work around this (just remove the /s) but unfortunately the last / is unpredictable. For example, if you call genDBUrl(host, 'blah/') then the output is /httpkey://127.0.0.1:9898/rsapublickey:0.0/test/blah/.

When XHR gets a URL like that what is it supposed to do? It can be safe in assuming the first / is an error, but what about the second?

The problem, btw, actually gets worse over time. Right now we only support absolute httpkey URLs with our custom XHR. But once we implement [issue 42](https://thali.codeplex.com/workitem/42) we will load the HTML page from a httpkey URL in which case having a relative URL path to a DB will make sense. So we can't just assume the first / is an error. 

So it seems to me that we should have code that knows that we are using an adapter and checks if the URL starts with that adapter's prefix. If it does then it's remote.

Thoughts?
